crates/primitives/simulations/src/lib.rs

```diff
@@ -51,6 +51,9 @@ impl From<Vec<SimulationFlagForEstimateFee>> for SimulationFlags {
             }
         }
 
+        // estimate_fee does not charge fees or do any balance checks
+        flags_out.charge_fee = false;
```

Comment:
Ya I know it's a little strange but I tried on sepolia

- running estimate_fee with a transfer txn which transfers 100% of the money in my wallet and it worked. if it tried to transfer fees at the end, it should've failed
- comparing the gas_consumed (not price) returned from estimate_fee and actual gas consued, actual one was higher
- starknet js multiplies estimate from the RPC by 1.5 for safety marging which probably causes it to work
- this is still a little akward and the behaviour I tested was for 0.5.1. It's a little complicated to check in the latest RPC because the pricing mechanism has changed, I have asked the Juno folks about it, waiting for an udpate.

Reply:
Just checked the Juno code, it does seem to skip fee charge - https://github.com/NethermindEth/juno/blob/a85cae7bc73fe34cfdd314276dae4596230d39d2/rpc/estimate_fee.go#L68

Reply:
That's a weeeeeeird ass design!